### PR TITLE
Unit test for Magento\Reports\Observer\EventSaver

### PR DESCRIPTION
**Pull request by karyna-t**
### Description (*)
Covers Magento\Reports\Observer\EventSaver class with a unit test.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Reports/Test/Unit/Observer/EventSaverTest.php [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+                ->method('getCustomerId')
+                ->willReturn(1);
+        } else {
+            $this->customerSessionMock->expects($this->once())
+                ->method('isLoggedIn')
+                ->willReturn(false);
+            $this->customerVisitorMock->expects($this->once())
+                ->method('getId')
+                ->willReturn(2);
+        }
```
Comment (lbajsarowicz) [inline app/code/Magento/Reports/Test/Unit/Observer/EventSaverTest.php]: Split that into 2 separate tests:
* `testSaveWithoutSubjectWhenLoggedIn`
* `testSaveWithoutSubjectForLoggedIn`

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
DataProvider is completely unnecessary there.

File: app/code/Magento/Reports/Test/Unit/Observer/EventSaverTest.php (hunk 1 above)
Comment (karyna-t) [reply app/code/Magento/Reports/Test/Unit/Observer/EventSaverTest.php]: thanks for the comment, done.

File: app/code/Magento/Reports/Test/Unit/Observer/EventSaverTest.php [hunk 2]
```diff
@@ -116,41 +116,30 @@ public function testSaveWithSubject()
...
-            [true],
-            [false]
-        ];
+        $this->customerSessionMock->expects($this->once())
+            ->method('isLoggedIn')
+            ->willReturn(false);
+        $this->customerVisitorMock->expects($this->once())
+            ->method('getId')
+            ->willReturn(2);
+        $this->eventSaver->save(1, 1, null);
```
Comment (lbajsarowicz) [inline app/code/Magento/Reports/Test/Unit/Observer/EventSaverTest.php]: Get rid of magic numbers, replace them with their meaning:
- `const STUB_EVENT_TYPE_ID`
- `const STUB_OBJECT_ID`

File: app/code/Magento/Reports/Test/Unit/Observer/EventSaverTest.php [hunk 3]
```diff
@@ -116,41 +116,30 @@ public function testSaveWithSubject()
...
-        return [
-            [true],
-            [false]
-        ];
+        $this->customerSessionMock->expects($this->once())
+            ->method('isLoggedIn')
+            ->willReturn(false);
+        $this->customerVisitorMock->expects($this->once())
+            ->method('getId')
+            ->willReturn(2);
```
Comment (lbajsarowicz) [inline app/code/Magento/Reports/Test/Unit/Observer/EventSaverTest.php]: Also that would be just show  that it's `const STUB_VISITOR_ID` (if I assume properly)

File: app/code/Magento/Reports/Test/Unit/Observer/EventSaverTest.php [hunk 4]
```diff
@@ -116,41 +116,30 @@ public function testSaveWithSubject()
...
-            $this->customerVisitorMock->expects($this->once())
-                ->method('getId')
-                ->willReturn(2);
-        }
+        $this->customerSessionMock->expects($this->once())
+            ->method('isLoggedIn')
+            ->willReturn(true);
+        $this->customerSessionMock->expects($this->once())
+            ->method('getCustomerId')
+            ->willReturn(1);
```
Comment (lbajsarowicz) [inline app/code/Magento/Reports/Test/Unit/Observer/EventSaverTest.php]: `const STUB_CUSTOMER_ID`

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Just get rid of Magic numbers and we're done!

File: app/code/Magento/Reports/Test/Unit/Observer/EventSaverTest.php (hunk 2 above)
Comment (karyna-t) [reply app/code/Magento/Reports/Test/Unit/Observer/EventSaverTest.php]: Thank you, fixed.

**Review by lbajsarowicz — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-6709](https://jira.corp.magento.com/browse/ENGCOM-6709) has been created to process this Pull Request
:eight_spoked_asterisk: @lbajsarowicz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**